source/fundamentals/bson/utf8-validation.txt

```diff
@@ -25,9 +25,10 @@ processing overhead since it needs to check the data.
 If you *disable* validation, your application avoids the validation processing
 overhead, but cannot guarantee consistent presentation of invalid UTF-8 data.
 
-The driver enables UTF-8 validation by default. It checks documents for any
-characters that are not encoded in a valid UTF-8 format when it transfers data
-between your application and MongoDB.
+By default, the driver enables UTF-8 validation on BSON strings in documents
+from MongoDB. It checks incoming strings for any characters that are not 
+encoded in a valid UTF-8 format when it transfers data from MongoDB to your 
+application.
```

Comment:
The section below is outside of the diff but I think we should take another look at improving its clarity if that's alright:

> The current version of the {+driver-short+} automatically substitutes
   invalid UTF-8 characters with alternate valid UTF-8 ones before
   validation when you send data to MongoDB. Therefore, the validation
   only throws an error when the setting is enabled and the driver
   receives invalid UTF-8 document data from MongoDB.

Specifically this bit:

> automatically substitutes invalid UTF-8 characters with alternate valid UTF-8 ones before validation when you send data to MongoDB.

I do not think we need to write a whole lesson on UTF and all the glossary of terms that come with it we can lean on the documentation here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#utf-16_characters_unicode_code_points_and_grapheme_clusters

The important bit from that is strings in JS are made up of UTF-16 code points and you sometimes need more than one code point to represent a character, when multi-code-points (surrogate pairs) are accidentally split up, there is no way to represent that character anymore so you cannot translate it to UTF-8. There is a well-known filler character that is used in its place, U+FFFD that makes the output UTF-8 valid without removing the character entirely. 

We should say "substitutes lone surrogates with the replacement character", ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/isWellFormed

> with alternate valid UTF-8 ones

It is specifically the replacement character which is U+FFFD

Reply:
used ref https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toWellFormed instead since this references U+FFFD and corrects invalid characters as opposed to strictly checking for them. Let me know if this is incorrect!